trieste/acquisition/rule.py

```diff
@@ -1282,7 +1282,19 @@ def state_func(
                     _points.append(rule.acquire(subspace, _models, _datasets))
                 points = tf.stack(_points, axis=1)
             else:
-                points = self._rule.acquire(acquisition_space, models, datasets)
+                # Filter out local datasets as this is the EGO rule with normal acquisition
```

Comment:
The fact that `self._rules` implies (EGO + no local models) isn't obvious here. Maybe rename `._rules` to 
`._subspace_rules` and change the initialisation (above) to:

```python
        if self._rules is None and not (
            _num_local_models == 0 and isinstance(self._rule, EfficientGlobalOptimization)
        ):
```

?

Reply:
I am not sure about renaming to `_subspace_rules`, but an happy to apply De Morgan's law to the condition expression.

Also, I'll change the comment slightly to be not so definitive about being EGO. That is the case for now, but may support more rules in the future.